scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1953,25 +1966,32 @@ private void sendMessage(final long nanos, final RawData message, final Connecti
 				initializeHandshaker(handshaker);
 				handshaker.startHandshake();
 			}
+			message.onConnecting();
 			handshaker.addApplicationDataForDeferredProcessing(message);
 		}
 		else {
+			boolean full = DtlsEndpointContext.HANDSHAKE_MODE_FORCE_FULL.equals(handshakeMode);
+			boolean force = full || DtlsEndpointContext.HANDSHAKE_MODE_FORCE.equals(handshakeMode);
 			Long timeout = autoResumptionTimeoutMillis;
-			String contextTimeout = message.getEndpointContext().get(DtlsEndpointContext.KEY_RESUMPTION_TIMEOUT);
-			if (contextTimeout != null) {
-				if (contextTimeout.isEmpty()) {
-					timeout = null;
-				} else {
-					try {
-						timeout = Long.valueOf(contextTimeout);
-					} catch (NumberFormatException e) {
+			if (force) {
+				connection.setResumptionRequired(true);
+			} else {
+				String contextTimeout = message.getEndpointContext().get(DtlsEndpointContext.KEY_RESUMPTION_TIMEOUT);
+				if (contextTimeout != null) {
+					if (contextTimeout.isEmpty()) {
+						timeout = null;
+					} else {
+						try {
+							timeout = Long.valueOf(contextTimeout);
+						} catch (NumberFormatException e) {
+						}
 					}
 				}
 			}
 			if (connection.isAutoResumptionRequired(timeout)) {
 				// create the session to resume from the previous one.
 				if (serverOnly) {
-					message.onError(new EndpointUnconnectedException("server only, resumption required!"));
+					message.onError(new EndpointUnconnectedException("server only, resumption requested failed!"));
```

Comment:
I'm a bit lost here.
We are in the else block which begins line 1972.
In this block `session != null || ticket != null`.
What does it mean exactly ?
1. We can have a connection with a sesson established ?
2. Is it also possible to have a connection linked to an existing session (maybe from cache) but with no session established ?

Should we launch an handshake when HANDSHAKE_NONE is used for 1. or 2.  and autoresumtion is required?

Reply:
```
DTLSSession session = connection.getEstablishedSession();
SessionTicket ticket = connection.getSessionTicket();
```
`session != null` => established
`ticket != null` => from cache

And sorry, I'm just lost by your questions.
FMPOV, I prefer to use "server only" instead of the NONE.
This is functionality, mainly requested by you. So, feel free to specify what you want.

Reply:
> This is functionality, mainly requested by you. So, feel free to specify what you want.

At short term we will use the NONE feature , but for now we don't plan to use autoresumption  neither session cache. So I just try to think what should be the API :-)

So, I try to be clearer.  
In case where we use both (server and client) mode and HANDSHAKE NONE  and we either have: 
1. a connection established and an autoresumption required,
2. or a session from cache (so no established session),

I ask myself if we should initialize a handshake. I would say no because if you say no handshake I suppose you don't want handshake at all but I'm not sure. :thinking:

Reply:
FMPOV, NONE is used to inhibit a handshake, so inhibit ...

Reply:
Hope, it's fixed now.

Reply:
:thinking: 

It seems to me that if ticket != null && session == null and handshake NONE.
We will not raise : `message.onError(new EndpointUnconnectedException("connection missing!"));`
And I'm not sure what will be the behavior, because we will go in 
```java
// session with peer has already been established,
// use it to send encrypted message
sendMessage(message, connection, session);
```
(and in this case we don't have session established)

Reply:
Next try ... :-)